### PR TITLE
US-1054 more useful message on tree load 404.

### PR DESCRIPTION
**Pull request by laurgarn**

### CODE REVIEW

**Review by archendel — APPROVED** [review]

File: src/utils/rest-api.js [hunk 1]
```diff
@@ -774,11 +774,15 @@ export function fetchNetworkModificationTree(studyUuid) {
...
-    );
+    return backendFetch(url, { method: 'get' }).then((response) => {
+        if (response.ok) {
+            return response.json();
+        } else {
+            return response.json().then((text) => {
+                return Promise.reject(text);
+            });
+        }
+    });
```
Comment (AbdelHedhili) [inline src/utils/rest-api.js:785]: why this change ? It's exactly the same or i'm misssing something ?

File: src/components/study-container.js [hunk 2]
```diff
@@ -316,16 +316,26 @@ export function StudyContainer({ view, onChangeTab }) {
...
-                    },
-                })
-            )
+            .catch((errorMessage) => {
+                if (errorMessage.status === 404)
+                    displayErrorMessageWithSnackbar({
+                        errorMessage: '',
+                        enqueueSnackbar: enqueueSnackbar,
+                        headerMessage: {
+                            headerMessageId: 'StudyAwfulStateRecreate',
```
Comment (AbdelHedhili) [inline src/components/study-container.js:325]: not a big fan of the naming "awful" but whatever

**Review by AbdelHedhili — CHANGES_REQUESTED** [review]

File: src/utils/rest-api.js (hunk 1 above)
Comment (laurgarn) [reply src/utils/rest-api.js:785]: Not exactly. For "exception path", content was awaited as flat string and is in fact json, as seen in browser console. This json is to be exploited in the caller.

File: src/components/study-container.js (hunk 2 above)
Comment (laurgarn) [reply src/components/study-container.js:325]: Not very proud of, but it is internal. If you are inspired, feel free to propose another name ;-)

Comment (archendel) [reply src/components/study-container.js:325]: StudyUnrecoverableStateRecreate could be a better property name

Comment (AbdelHedhili) [reply src/components/study-container.js:325]: +1

**Review by AbdelHedhili — APPROVED** [review]

**Comment by SlimaneAmar** [comment]
Use `useSnackMessage`

**Pull request merged**